waspc/data/Generator/templates/sdk/wasp/server/auth/hooks.ts

```diff
@@ -0,0 +1,79 @@
+import type { Request as ExpressRequest } from 'express'
+import type { ProviderId, createUser } from '../../auth/utils.js'
+import { prisma } from '../index.js'
+import { Expand } from '../../universal/types.js'
+
+export type OnBeforeSignupHook = (
+  params: Expand<OnBeforeSignupHookParams>,
+) => void | Promise<void>
+
+export type OnAfterSignupHook = (
+  params: Expand<OnAfterSignupHookParams>,
+) => void | Promise<void>
+
+/**
+ * @returns Object with a URL that the OAuth flow should redirect to.
+ */
+export type OnBeforeOAuthRedirectHook = (
+  params: Expand<OnBeforeOAuthRedirectHookParams>,
+) => { url: URL } | Promise<{ url: URL }>
+
+// NOTE: We should be exporting types that can be reached by users via other
+// exported types (e.g. using the Parameters<T> Typescript helper).
+// However, we are not exporting this type to keep the API surface smaller.
+// This type is only used internally by the SDK. Exporting it might confuse
+// users since the name is too similar to the exported function type.
+// Same goes for all other *Params types in this file.
+type OnBeforeSignupHookParams = {
+  /**
+   * Provider ID object that contains the provider name and the provide user ID.
+   */
+  providerId: ProviderId
+} & CommonAuthHookParams
+
+type OnAfterSignupHookParams = {
+  /**
+   * Provider ID object that contains the provider name and the provide user ID.
+   */
+  providerId: ProviderId
+  /**
+   * User object that was created during the signup process.
+   */
+  user: Awaited<ReturnType<typeof createUser>>
+  oauth?: {
+    /**
+     * Access token that was received during the OAuth flow.
+     */
```

Comment:
My personal feeling is that most of these type field comments are redundant and just repeat the information visible from the code.

Examples include:
```
oauth.accessToken - access token received during the oauthFlow (redundant)
url - URL that the OAuth flow should redirect to (not redundant, but 
perhaps better to call the field `oauthRedirectUrl` and drop the comment)
hookName - Name of the hook that is being executed (redundant)
```
I'm getting Cupic JavaDoc vibes :smile:

Reply:
These comments are for the end user like some sort of documentation they can see at a glance: 
<img width="617" alt="Screenshot 2024-06-10 at 12 33 59" src="https://github.com/wasp-lang/wasp/assets/2223680/031dcd64-240a-4519-ba8f-cd3bf3a07164">

From a point of view of a developer developing Wasp (us) they are too much, but from the point of view of a Wasp user they seem okay to me. Giving a bit of extra "context". `Access token that was received during the OAuth flow.` mentions where the access token came from and might help them with understanding the flow better.

With that in mind, of course we can improve the comments to be less redundant. But do you agree with my intention here?

Reply:
Ohh, so they do show up. It never occurred to me to check the individual properties, I was only hovering on the value.

Yes, I agree wholeheartedly.